storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -304,23 +304,25 @@ public List<SignedBeaconBlock> getNonCanonicalBlocksAtSlot(final UInt64 slot) {
 
   protected void storeFinalizedBlocksToDao(
       final Collection<SignedBeaconBlock> blocks,
-      final Map<UInt64, List<BlobSidecar>> finalizedBlobSidecarsBySlot) {
+      final Map<SlotAndBlockRoot, List<BlobSidecar>> finalizedBlobSidecarsBySlot,
+      final Optional<UInt64> maybeEarliestBlobSidecar) {
     try (final FinalizedUpdater updater = finalizedUpdater()) {
       blocks.forEach(
           block -> {
             updater.addFinalizedBlock(block);
             // If there is no slot in BlobSidecar's map it means we are pre-Deneb or not in
             // availability period
-            if (!finalizedBlobSidecarsBySlot.containsKey(block.getSlot())) {
+            if (!finalizedBlobSidecarsBySlot.containsKey(block.getSlotAndBlockRoot())) {
               return;
             }
-            final List<BlobSidecar> blobSidecars = finalizedBlobSidecarsBySlot.get(block.getSlot());
-            if (blobSidecars.isEmpty()) {
-              updater.addNoBlobsSlot(block.getSlotAndBlockRoot());
-            } else {
-              blobSidecars.forEach(updater::addBlobSidecar);
-            }
+            finalizedBlobSidecarsBySlot
+                .get(block.getSlotAndBlockRoot())
+                .forEach(updater::addBlobSidecar);
           });
+      final Optional<UInt64> maybeEarliestBlobSidecarSlotDb = dao.getEarliestBlobSidecarSlot();
+      if (maybeEarliestBlobSidecarSlotDb.isEmpty() && maybeEarliestBlobSidecar.isPresent()) {
+        updater.setEarliestBlobSidecarSlot(maybeEarliestBlobSidecar.get());
+      }
       updater.commit();
```

Comment:
what about

```
maybeEarliestBlobSidecar.ifPresent(earliestBlobSidecar -> {
if(dao.getEarliestBlobSidecarSlot().isEmpty()) {
updater.setEarliestBlobSidecarSlot(earliestBlobSidecar);
}
});
```

so we query only if need?

Reply:
I've changed to the same idea but I like this code more
```
      if (maybeEarliestBlobSidecar.isPresent() && dao.getEarliestBlobSidecarSlot().isEmpty()) {
        updater.setEarliestBlobSidecarSlot(maybeEarliestBlobSidecar.get());
      }
```